packages/design-system/src/components/web-components/ds-text-field/ds-text-field.tsx

```diff
@@ -0,0 +1,67 @@
+import { define } from '../preactement/define';
+import { TextField, TextFieldProps } from '../../TextField';
+import { parseBooleanAttr } from '../wrapperUtils';
+import { formAttrs } from '../shared-attributes/form';
+import { PHONE_MASK, SSN_MASK, ZIP_MASK, CURRENCY_MASK } from '../../TextField/useLabelMask';
+
+const attributes = [
+  // Using the lowercase HTML attribute name rather than `auto-focus` so it's
+  // more natural. There's no reason for us to worry about name collisions.
+  'autofocus',
+  'aria-disabled',
+  'class-name',
+  'default-value',
+  'disabled',
+  'field-class-name',
+  'inversed',
+  'label-mask',
+  'multiline',
+  'name',
+  'numeric',
+  'value',
+  'root-id',
+  'size',
+  'type',
+  ...formAttrs,
+];
+
+function getMaskFunction(value: string) {
+  switch (value) {
+    case 'phone':
+      return PHONE_MASK;
+    case 'zip':
+      return ZIP_MASK;
+    case 'ssn':
+      return SSN_MASK;
+    case 'currency':
+      return CURRENCY_MASK;
+  }
+  return undefined;
+}
+
+interface WrapperProps
+  extends Omit<
+    TextFieldProps,
+    'autoFocus' | 'ariaDisabled' | 'disabled' | 'inversed' | 'labelMask'
+  > {
+  autofocus?: string;
+  ariaDisabled?: string;
+  disabled?: string;
+  inversed?: string;
+  labelMask?: string;
+  rootId?: string;
+}
+
+const Wrapper = ({ rootId, ...otherProps }: WrapperProps) => (
+  <TextField
+    {...otherProps}
```

Comment:
Are you talking about collisions with the props we defined below on lines 58-63?

Reply:
Yes, sorry I should've used one of those props explicitly. Like if otherprops contains `autofocus` what happens? This should also be considered a nit, and a non-blocking q.

Reply:
In that case, it would get overwritten by our autofocus prop that we explicitly define, and that's intentional, because we want to pass a parsed boolean to the Preact component. Except in this case there's a difference between `autofocus` and `autoFocus`, isn't there? So maybe this particular prop/attribute has a bug, and we need to destructure `autocomplete` out of `otherProps.

Reply:
Fixed in [54c0cad](https://github.com/CMSgov/design-system/pull/3172/commits/54c0cad7cdb562f0dd88216b76b3d2a88f29b5ef)